### PR TITLE
[TEP-0096]: Remove TEP-0074 Features as blockers

### PR DESCRIPTION
**Pull request by lbernick**
This commit proposes removing any new features in TEP-0074 as V1 blockers.
This does not mean we are stopping work on or deprioritizing these features,
but it does mean that we can release a V1 API before they are implemented.

### CODE REVIEW

**Comment by lbernick** [comment]
@bobcatfish @jerop @afrittoli @vdemeester @pritidesai please let me know your thoughts.

**Review by vdemeester — COMMENTED** [review]
Hum 🤔 What does this mean though ? We still do want to deprecate PipelineResource for v1 (meaning they won't be part of the API) right or not ? Is this stating we don't want to block on that decision and we could end up with PipelineResource in v1 ?

**Comment by lbernick** [comment]
> Hum 🤔 What does this mean though ? We still do want to deprecate PipelineResource for v1 (meaning they won't be part of the API) right or not ? Is this stating we don't want to block on that decision and we could end up with PipelineResource in v1 ?

I think we do not want to have PipelineResources in V1 (the TEP states "remove the `Resources` fields of `Task`, `TaskRun`, `Pipeline`, and `PipelineRun`"). TEP-0074 is really valuable for laying out the features that PipelineResources provided that we should replace, but I don't think it makes sense to block V1 on new experimental features. This means IMO it would be OK to release V1 without PipelineResources, and without having completed the features in TEP-0074.

In terms of a migration path for PipelineResource users: we do have all the functionality available in Tekton to express the same workflows as PipelineResources allowed us to; it seems like a question of how easy it is (e.g. do you run into scaling problems, how verbose are your pipeline definitions). My understanding is that IBM was able to move off of PipelineResources, Google isn't using them, (not sure about Red Hat) and we haven't heard from any users saying that they would be unable to migrate to a V1 API without the features in TEP-0074 (although in fairness my survey got very few responses). It's a difficult judgment call, but IMO these features fall under "things we are actively working on to improve Tekton" but not "things that are required for a functional complete product".

**Comment by vdemeester** [comment]
Ok fair enough, as long as we remove `PipelineResource` in v1, I am good 😛

**Comment by bobcatfish** [comment]
I'm fine with this change - I think it requires a few changes in TEP-0074 as well - looking at the proposal https://github.com/tektoncd/community/blob/main/teps/0074-deprecate-pipelineresources.md#proposal step 8:

> Once this TEP is implementable, and we create v1 types, we would [remove] any references to PipelineResources

This implies that we would need TEP-0074 to be implementable before we'd create v1 types that don't reference PipelineResources - and https://github.com/tektoncd/community/blob/main/teps/0074-deprecate-pipelineresources.md#requirements-to-mark-as-implementable requires TEP-0044 before marking as implementable

There are a few other references to v1 that it might make sense to update as well

> we do have all the functionality available in Tekton to express the same workflows as PipelineResources allowed us to

i'm nit picking but while it is possible to express the same workflows, the same functionality still isn't available without something that provides the functionality "Augmenting Tasks with steps that execute on the same pod"

/approve

but i think we need the TEP-0074 changes to go in as well so adding a hold for that (so we aren't inconsistent across these 2 teps)

/hold

**Comment by lbernick** [comment]
> I'm fine with this change - I think it requires a few changes in TEP-0074 as well

Done in #641.

> i'm nit picking but while it is possible to express the same workflows, the same functionality still isn't available without something that provides the functionality "Augmenting Tasks with steps that execute on the same pod"

Agreed, makes sense!

**Review by jerop — APPROVED** [review]
thanks @lbernick!

@tektoncd/core-maintainers - we need a non-googler approver here 
(@vdemeester are you available to review/approve since you already reviewed? 😁)

**Comment by pritidesai** [comment]
yup, I agree with @bobcatfish, `pipelineResources` provides the functionality which cannot be directly substituted with the existing API but not a V1 blocker 🎉 

/approve

For API WG on 3/14, this is ready to merge 👍

**Comment by jerop** [comment]
it was approved for merging in API WG on 3/7 (cc @pritidesai)

/lgtm

**Pull request merged**
